extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/handler/BridgeHandler.java

```diff
@@ -0,0 +1,643 @@
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link BridgeHandler} is the handler for a digitalSTROM-Server and connects it to
+ * the framework.<br>
+ * All {@link DeviceHandler}s and {@link SceneHandler} use the {@link BridgeHandler} to execute the actual
+ * commands.
+ * <p>
+ * The {@link BridgeHandler} also:
+ * <ul>
+ * <li>manages the {@link DeviceStatusManager} (starts, stopps, register {@link DeviceStatusListener},
+ * register {@link SceneStatusListener} and so on)</li>
+ * <li>starts the {@link DsThingTypeProvider} if the connection is established</li>
+ * <li>load the configurations for the {@link Config}.</li>
+ * <li>implements {@link ManagerStatusListener} to manages the expiration of the Thing initalisations</li>
+ * <li>implements the {@link ConnectionListener} to manages the {@link ThingStatus} of all connected Things
+ * </li>
+ * <li>and implements the {@link TotalPowerConsumptionListener} to uptate his Channels.</li>
+ * </ul>
+ * </p>
+ *
+ * @author Michael Ochel - Initial contribution
+ * @author Matthias Siegele - Initial contribution
+ */
+public class BridgeHandler extends BaseBridgeHandler
+        implements ConnectionListener, TotalPowerConsumptionListener, ManagerStatusListener {
+
+    private Logger logger = LoggerFactory.getLogger(BridgeHandler.class);
+
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_DSS_BRIDGE);
+
+    /* DS-Manager */
+    private ConnectionManager connMan;
+    private StructureManager structMan;
+    private SceneManager sceneMan;
+    private DeviceStatusManager devStatMan;
+
+    private List<SceneStatusListener> sceneListener;
+    private List<DeviceStatusListener> devListener;
+    private DsThingTypeProvider thingTypeProvider = null;
+
+    private class InitalizazeRunnable implements Runnable {
+
+        BridgeHandler bridge = null;
+        Configuration configuration;
+        String[] loginConfig;
+
+        public InitalizazeRunnable(BridgeHandler bridge, Configuration configuration, String[] loginConfig) {
+            this.bridge = bridge;
+            this.configuration = configuration;
+            this.loginConfig = loginConfig;
+        }
+
+        @Override
+        public void run() {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_PENDING, "Loading configurations");
+            // get Configurations
+            if (configuration.get(DigitalSTROMBindingConstants.SENSOR_DATA_UPDATE_INTERVAL) != null
+                    && !configuration.get(DigitalSTROMBindingConstants.SENSOR_DATA_UPDATE_INTERVAL).toString().trim()
+                            .replace(" ", "").isEmpty()) {
+
+                Config.SENSORDATA_REFRESH_INTERVAL = Integer.parseInt(
```

Comment:
Config must NOT be static! The configuration is specific to this dSS. You can potentially have multiple dSSs in your installation with different configurations. So you need an individual instance of Config here.